src/nl/hannahsten/texifyidea/refactoring/inlinefile/LatexInlineFileHandler.kt

```diff
@@ -0,0 +1,125 @@
+import com.intellij.openapi.ui.DialogWrapper
+import com.intellij.psi.PsiElement
+import com.intellij.psi.util.PsiUtilBase.getElementAtCaret
+import nl.hannahsten.texifyidea.LatexLanguage
+import nl.hannahsten.texifyidea.file.LatexFile
+import nl.hannahsten.texifyidea.psi.LatexCommands
+import nl.hannahsten.texifyidea.reference.InputFileReference
+import nl.hannahsten.texifyidea.util.Log
+import nl.hannahsten.texifyidea.util.appendExtension
+import nl.hannahsten.texifyidea.util.files.isLatexFile
+import nl.hannahsten.texifyidea.util.firstParentOfType
+
+/**
+ * Allows for inlining an include command. Does not do the inlining itself.
+ *
+ * @author jojo2357
+ */
+class LatexInlineFileHandler : InlineActionHandler() {
+
+    override fun isEnabledForLanguage(language: Language?): Boolean {
+        return language == LatexLanguage
+    }
+
+    override fun canInlineElement(element: PsiElement?): Boolean {
+        return canInlineLatexElement(element)
+    }
+
+    override fun inlineElement(project: Project, editor: Editor?, element: PsiElement) {
+        // Resolve the file to be inlined
+        val inlineFile: LatexFile = resolveInlineFile(element) ?: return
+
+        val dialog = LatexInlineFileDialog(
+            project,
+            inlineFile,
+            getReference(element, editor),
+            editor != null
+        )
+
+        if (dialog.myOccurrencesNumber > 0) {
+            if (ApplicationManager.getApplication().isUnitTestMode) {
+                try {
+                    dialog.doAction()
+                }
+                finally {
+                    dialog.close(DialogWrapper.OK_EXIT_CODE, true)
+                }
+            }
+            else {
+                dialog.show()
+            }
+        }
+        else {
+            Notification(
+                "LaTeX",
+                "No usages found",
+                "Could not find any usages for ${inlineFile.name}",
+                NotificationType.ERROR
+            ).notify(project)
+        }
+    }
+}
+
+/**
+ * This is static only so that the unit tests can use it, and also since it can be static
+ */
+fun canInlineLatexElement(element: PsiElement?): Boolean {
+    val out = when (element) {
+        is LatexFile -> {
+            element.containingFile.isLatexFile()
+        }
+
+        is LatexCommands -> {
+            (element.references.filterIsInstance<InputFileReference>().isNotEmpty())
+        }
+
+        else -> true
+    }
+
+    return out
+}
+
+fun resolveInlineFile(element: PsiElement) = when (element) {
+    // If we tried to inline the file in file view, or the argument in `input`
+    is LatexFile -> element
+    // If the caret was on the `input` command
+    is LatexCommands -> {
+        val ifr = element.references.filterIsInstance<InputFileReference>()[0]
+
+        element.containingFile.parent?.findFile(
+            ifr.key.appendExtension(ifr.defaultExtension)
+        ) as? LatexFile
```

Comment:
You can replace this by `ifr.resolve(lookForInstalledPackages = false)`, which is much more likely to work!

Reply:
Running tests now

Reply:
I could inline `ifr` now :thinking: